util/internal/test/src/main/java/org/hibernate/search/util/impl/test/rule/log4j/LogProgrammatic.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.util.impl.test.rule.log4j;
+
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.core.Appender;
+import org.apache.logging.log4j.core.LoggerContext;
+import org.apache.logging.log4j.core.config.Configuration;
+import org.apache.logging.log4j.core.config.LoggerConfig;
+
+public class LogProgrammatic {
```

Comment:
This is a really unusual class name... Maybe rename to something more "to the point", such as `Log4j2AppenderHelper` or `Log4j2ConfigurationAccessor`?